docs/graasp-insights-user-guide.md

```diff
@@ -0,0 +1,243 @@
+
+Add an algorithm from your file system. Although any `python` algorithm can be added, dataset loading and saving as well as parameter handling is done in a specific way. It is therefore recommended having a look at existing algorithms to have a better understanding of how the algorithm should communicate with the interface. The `Write algorithm` option is also a good alternative to quickly learn about what the algorithm layout should look like.
+
+![Load algorithm from a file](./figures/load-algorithm-from-file.png)
+
+2. > Add Graasp algorithm
+
+Select a premade Graasp algorithm to add to the application. Those are already present when first launching Graasp Insights. If you delete a pre-existing algorithm, you can add it back with this option.
+
+![Add Graasp algorithm](./figures/add-graasp-algorithm.png)
+
+3. > Write algorithm
+
+An in-app editor is available for you to write your algorithms. The editor is filled with pre-written code taking care of handling parameters as well as loading and saving the dataset. Write your code in the designated `write your dataset changes here` area.
+
+![Write algorithm](./figures/write-algorithm.png)
+
+Give the algorithm a name and a description so that you can refer to it later when executing algorithms.
+
+Both anonymization and validation algorithms can be written and switching between the two can be done with the `Type` menu select.
+
+Click on the `SAVE` button when you are done writing your algorithm. The algorithm will then be available in the [Executions](#Executions) tab.
+
+## Adding parameters
+
+When adding an algorithm with the `Write algorithm` option, a new parameter can be added by clicking on the `ADD PARAMETER` button.
+
+Choose a type and enter a name, a description and a default value for your new parameter. As you modify the parameters, the code is automatically updated to accommodate for the new parameters. You can then access your parameter with `args.parameter_name`.
+
+![Adding parameters](./figures/add-algorithm-parameters.png)
+
+Four types are available for you to choose:
+
+- `Integer`
+- `Float` (Floating point number)
+- `String` (Sequence of characters)
+- `Field Selector`
+
+The `Field Selector` consists of a nested attribute selection. The user can select attributes of a dataset and that selection can be accessed within the algorithm code. Use the following code to load a field selection named `fields`:
+
+```python
+import json
+
+fields = json.loads(args.fields)
+```
+
+The `Field Selector` is particularly useful for executing algorithms on specific parts of a dataset. The attributes that are shown depend entirely on the schema. If there is more than one schema loaded in Graasp Insights, you can switch between schemas with the schema selection that shows up at the top of the `Parameters` section.
+
+> Note: if no schema corresponds to the dataset you want to apply the algorithm on, you can generate one for your dataset in the [Schemas](#Schemas) tab.
+
```

Comment:
Maybe let's have a note that a parameter cannot be empty otherwise you cannot save the algorithm... 
And names must be unique.